lib/utils.pm

```diff
@@ -778,14 +778,10 @@ sub disable_serial_getty {
     return if check_var('BACKEND', 's390x');
     # Stop serial-getty on serial console to avoid serial output pollution with login prompt
     # Doing early due to bsc#1103199 and bsc#1112109
-    # Return if already disabled
-    return if script_run "systemctl is-enabled $service_name";
-    systemctl "stop $service_name",    ignore_failure => 1;
-    systemctl "disable $service_name", ignore_failure => 1;
-    record_info 'serial-getty', "Serial getty disabled for $testapi::serialdev";
     # Mask if is qemu backend as use serial in remote installations e.g. during reboot
-    systemctl "mask $service_name", ignore_failure => 1 if check_var('BACKEND', 'qemu');
-    record_info 'serial-getty', "Serial getty mask for $testapi::serialdev";
+    my $cmd = check_var('BACKEND', 'qemu') ? 'mask' : 'disable';
+    systemctl is_sle('<12-sp3') ? "stop $service_name; systemctl $cmd $service_name" : "$cmd --now $service_name";
```

Comment:
This is the main concern I have. Similar to zypper_call, it will fail the test if fails. And as is executed everywhere, I would not take a risk to merge it straight away. Also I'm not sure about maintenance updates of Leap 42.3.
In general looks good.

Reply:
Leap 42.3 also supports the "--now" parameter. The exit code of "stop $service_name" is still ignored due to the ";", same as in before

Reply:
Cannot `disable_and_stop_service` be used here?

```suggestion
    $mask_service = check_var('BACKEND', 'qemu') ? 1 : 0;
    disable_and_stop_service('$service_name', mask_service => $mask_service);
```

Reply:
When `disable_and_stop_service` is already created and used then it would make sense to use it also here, depends if the `--now` feature need to be used.

Reply:
Yes, I might take a look into that. My PR was initially created before `disable_and_stop_service` was created